rfcs/0004-policy-sync-v1.md

```diff
@@ -0,0 +1,182 @@
+  relies on [OCM](https://open-cluster-management.io/) as a Multicluster solution, which defines its own unrelated
+  set of Policies and Policy Framework. Unless explicitely mentioned, this document
+  refers to Policies as Gateway API Policies.
+
+* Policy overriding: The concept of policy overriding is mentioned in this document. It refers to the proposed ability of the downstream Gateway implementation to prioritise downstream Policies against synced Policies in case of conflicts.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, Kuadrant's support for the Policy Attachment Model can be divided in
+two categories:
+* Policies targeting the Multicluster Gateway, defined in the hub cluster and
+  reconciled by the Multicluster Gateway Controller
+* Policies targeting the downstream Gateway, defined in the spoke clusters and
+  reconciled by the downstream Gateway controllers.
+
+In a realistic multicluster scenario where multiple spoke clusters are present, the management of these policies can become tedious and error-prone, as policies have
+to be defined in the hub cluster, as well as replicated in the multiple spoke clusters.
+
+As Kuadrant users:
+* Gateway-admin has a set of homogeneous clusters and needs to apply per cluster rate limits across the entire set.
+* Platform-admin with a set of clusters with rate limits applied needs to change rate limit for one particular cluster. 
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The policy sync feature will allow a gateway-admin to configure, via GatewayClass
+parameters, a set of Policy GVRs to be synced by the Multicluster Gateway Controller.
+
+The `policiesToSync` field in the parameters defines those GVRs. For example, in
+order to configure the controller to sync AuthPolicies:
+
+```json
+"policiesToSync": [
+  {
+    "group": "kuadrant.io",
+    "version": "v1beta1",
+    "resource": "authpolicies" 
+  }
+]
+```
+
+The support for resources that the controller can sync is limited by the following:
+* The controller ServiceAccount must have permission to watch, list, and get the
+  resource to be synced
+* The resource must implement the Policy schema:
+    * Have a `.spec.targetRef` field
+
+When a Policy is configured to be synced in a GatewayClass, the Multicluster
+Gateway Controller starts watching events on the resources, and propagates changes
+by placing the policy in the spoke clusters, with the following mutations:
+* The `TargetRef` of the policy is changed to reference the downstream Gateway
+* The `kuadrant.io/policy-synced` annotation is set
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### Process overview
+
+#### Dynamic Policy watches
+
+The Multicluster Gateway Controller reconciles parameters referenced by the
+GatewayClass of a Gateway. A new field is added to the parameters that allows
```

Comment:
Will I have to add that I want to sync AP/RLP to every gateway class I create?

Reply:
Not necessarily, as that is configured in a ConfigMap, it would be possible to have multiple gateway classes referencing the single ConfigMap so the configuration doesn't have to be repeated